libs/testserver/pipelines_test.go

```diff
@@ -0,0 +1,246 @@
+package testserver
+
+import (
+	"testing"
+
+	"github.com/databricks/databricks-sdk-go/service/pipelines"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestPipelineStartUpdate_StoresUpdateParameters(t *testing.T) {
+	workspace := NewFakeWorkspace("http://test")
+
+	pipelineId := "test-pipeline-123"
+	workspace.Pipelines[pipelineId] = pipelines.GetPipelineResponse{
+		PipelineId: pipelineId,
+		Name:       "Test Pipeline",
+	}
+
+	// Test case 1: Basic update with no parameters
+	req1 := Request{
+		Body: []byte(`{}`),
+	}
+
+	response1 := workspace.PipelineStartUpdate(req1, pipelineId)
+	require.Equal(t, 0, response1.StatusCode)
+
+	assert.Len(t, workspace.PipelineUpdates, 1)
+
+	startUpdateResponse, ok := response1.Body.(pipelines.StartUpdateResponse)
+	require.True(t, ok)
+	updateId1 := startUpdateResponse.UpdateId
+
+	storedUpdate1, exists := workspace.PipelineUpdates[updateId1]
```

Comment:
We should use PipelineGetUpdate. We can assert against the response. For example, what value should we see for "state" field? This way we will test using the API used by the client, instead of asserting on how we represent objects internally.